frontend/src/views/CalculateNutrients/FertilizerModal/NewFertilizerModal.tsx

```diff
@@ -0,0 +1,287 @@
+import Divider from '@mui/material/Divider';
+import Modal, { ModalProps } from '@/components/common/Modal/Modal';
+import { NMPFileFarmManureData } from '@/types/NMPFileFarmManureData';
+import { formCss, formGridBreakpoints } from '@/common.styles';
+import { APICacheContext } from '@/context/APICacheContext';
+import { InputField } from '@/components/common';
+
+type NutrientObject = { N: number; P: number; K: number };
+
+type FertilizerFormData = {
+  fieldName: string;
+  fertilizerType: number;
+  fertilizerId: number;
+  applicationRate: number;
+  applUnit: number;
+  liquidDensity: number;
+  densityUnits: number;
+  availableNutrients: NutrientObject;
+  nutrientsStillRequired: NutrientObject;
+};
+
+type FertilizerModalProps = {
+  initialModalData?: FertilizerFormData | undefined;
+  handleSubmit?: (data: NMPFileFarmManureData) => void;
+  onCancel: () => void;
+};
+
+const FERTILIZER_METHOD: Array<{ id: string; label: string }> = [
+  { id: 'Broadcast', label: 'Broadcast' },
+  { id: 'Banded', label: 'Banded' },
+  { id: 'With planter', label: 'With planter' },
+  { id: 'Sidedress', label: 'Sidedress' },
+  { id: 'Fertigation', label: 'Fertigation' },
+  { id: 'Follar', label: 'Follar' },
+];
+
+export default function FertilizerModal({
+  initialModalData,
+  // Import field and handleSubmit from parent component when ready
+  // field,
+  // handleSubmit,
+  onCancel,
+  ...props
+}: FertilizerModalProps & Omit<ModalProps, 'title' | 'children' | 'onOpenChange'>) {
+  const [fertilizerTypes, setFertilizerTypes] = useState<
+    {
+      id: number;
+      label: string;
+      name: string;
+      dryliquid: string;
+      custom: boolean;
+    }[]
+  >([]);
+  const [fertilizerOptions, setFertilizerOptions] = useState<
+    {
+      id: number;
+      label: string;
+      name: string;
+      dryliquid: string;
+      nitrogen: number;
+      phosphorous: number;
+      potassium: number;
+      sortnum: number;
+    }[]
+  >([]);
+  const [filteredFertilizersOptions, setFilteredFertilizerOptions] = useState<
+    {
+      id: number;
+      label: string;
+      name: string;
+      dryliquid: string;
+      nitrogen: number;
+      phosphorous: number;
+      potassium: number;
+      sortnum: number;
+    }[]
+  >([]);
+  const [fertilizerUnits, setFertilizerUnits] = useState<
+    {
+      id: number;
+      name: string;
+      dryliquid: string;
+      conversiontoimperialgallonsperacre: number;
+    }[]
+  >([]);
+
+  const [formData, setFormData] = useState<FertilizerFormData>(
+    initialModalData ?? {
+      fieldName: '',
+      fertilizerType: 0,
+      fertilizerId: 0,
+      applicationRate: 1,
+      applUnit: 1,
+      liquidDensity: 1,
+      densityUnits: 0,
+      availableNutrients: { N: 0, P: 0, K: 0 },
+      nutrientsStillRequired: { N: 0, P: 0, K: 0 },
+    },
+  );
+  const apiCache = useContext(APICacheContext);
+
+  const dryOrLiquidUnitOptions = fertilizerUnits.filter(
```

Comment:
What is this filter checking for?

Reply:
Dry and liquid fertilizers use different units. It filters out units not applicable to the dry/liquid fertilizer selected by the user, to be used in the drop down selector.